src/Admin/AbstractAdminExtension.php

```diff
@@ -96,6 +96,15 @@ public function configureBatchActions(AdminInterface $admin, array $actions): ar
         return $actions;
     }
 
+    // NEXT_MAJOR: Remove the PHPDoc block as the interface will then specify the types
+    /**
+     * @param mixed[] $idx
+     * @phpstan-param AdminInterface<T> $admin
+     */
+    public function preBatchAction(AdminInterface $admin, string $actionName, ProxyQueryInterface $query, array &$idx, bool $allElements = false): void
```

Comment:
Do we need the default value for `allElements`? There is no call without that argument.

Reply:
This is consistent with https://github.com/sonata-project/SonataAdminBundle/blob/ea8c6ac2d8eff155447477c170cf6d8db7a8d623/src/Admin/AdminInterface.php#L254

Reply:
There is also no call for the `AdminInterface::preBatchAction` that does not provide the `allElements` parameter:
https://github.com/sonata-project/SonataAdminBundle/blob/50ef7e152413c16ebdaacff112ea09877141d6c8/src/Controller/CRUDController.php#L471

It looks more like some kind of *bug*, that we provide a default value here. We should ignore the current consistency in favor of advertising the correct arguments.

Reply:
Can be confusing for the user to have two different method signatures.

What about keeping it like this with a NEXT MAJOR on both method to remove the default argument ?

Reply:
> Can be confusing for the user to have two different method signatures.
> 
> What about keeping it like this with a NEXT MAJOR on both method to remove the default argument ?

This can be done on the stable branch
https://3v4l.org/V0cUU

Reply:
Indeed @7ochem can you do it ?

Reply:
Just so I understand this correctly:

I remove the default from the interface?
```diff
- * @method void preBatchAction(AdminInterface $admin, string $actionName, ProxyQueryInterface $query, array &$idx, bool $allElements = false)
+ * @method void preBatchAction(AdminInterface $admin, string $actionName, ProxyQueryInterface $query, array &$idx, bool $allElements)
```

But keep it on the AbstractAdminExtension implementation, here:
https://github.com/sonata-project/SonataAdminBundle/blob/7776c37c3e6abdbac5c717d7fdcea0a50484f02d/src/Admin/AbstractAdminExtension.php#L104

Am I correct @core23 ?

Reply:
I think he want to remove it in both the Interface and the implementation (and remove it too for the AdminINterface)

Reply:
> I think he want to remove it in both the Interface and the implementation

👍🏻 Ok, I'll change that and push it

> (and remove it too for the AdminINterface)

Removing it inside the existing method in the AdminInterface is a BC break. Alos, in my opinion, that would be outside of the scope of this PR. However I would be happy to issue a new PR afterwards for this change.